lib/Cmt.ml

```diff
@@ -29,148 +31,138 @@ end
 
 include T
 include Comparator.Make (T)
+open Fmt
 
-let create txt loc = {txt; loc}
+type pos = Before | Within | After
 
-let split_asterisk_prefixed {txt; loc= {Location.loc_start; _}} =
-  let len = Position.column loc_start + 3 in
-  let pat =
-    String.Search_pattern.create
-      (String.init len ~f:(function
-        | 0 -> '\n'
-        | n when n < len - 1 -> ' '
-        | _ -> '*' ) )
-  in
-  let rec split_asterisk_prefixed_ pos =
-    match String.Search_pattern.index pat ~pos ~in_:txt with
-    | Some 0 -> "" :: split_asterisk_prefixed_ len
-    | Some idx ->
-        String.sub txt ~pos ~len:(idx - pos)
-        :: split_asterisk_prefixed_ (idx + len)
-    | _ ->
-        let drop = function ' ' | '\t' -> true | _ -> false in
-        let line = String.rstrip ~drop (String.drop_prefix txt pos) in
-        if String.is_empty line then [" "]
-        else if Char.equal line.[String.length line - 1] '\n' then
-          [String.drop_suffix line 1; ""]
-        else if Char.is_whitespace txt.[String.length txt - 1] then
-          [line ^ " "]
-        else [line]
-  in
-  split_asterisk_prefixed_ 0
+module Asterisk_prefixed = struct
```

Comment:
@gpetiot in case you don't like this name, these comments are also known as "banner" comments due to how they hang off a pole (of asterisks).